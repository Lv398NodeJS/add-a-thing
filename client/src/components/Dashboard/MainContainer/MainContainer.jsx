import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container, Row, Col } from 'react-bootstrap';
import { bindActionCreators } from 'redux';
// import { getTaskRef } from '../TaskItem/TaskItemUtils';
import * as mainContainer from '../../../actions/mainContainerActions';
import TasksColumn from '../TasksColumn/TasksColumn';
import MainInput from '../MainInput/MainInput';
// import { deleteDragByEvent, handleDeleteDropCSS } from './MainContainerUtils';
import './MainContainer.scss';
import trash from '../../assets/trash.svg';

class MainContainer extends Component {
  componentWillMount() {
    const dashboardId = document.URL.split('/').pop();
    const { mainContainerActions } = this.props;
    // mainContainerActions.setTaskListRef(taskListRef);
    mainContainerActions.fetchTaskList(dashboardId);
  }

  // handleTaskDrop = (taskID, newStatus) => {
  //   const { taskList, taskListRef } = this.props;
  //   const [taskData] = taskList.filter(task => task.id === taskID);

  //   if (taskData && taskData.status !== newStatus) {
  //     const updatedTask = {
  //       name: taskData.name,
  //       status: newStatus,
  //       priority: taskData.priority || 'Medium',
  //       description: taskData.description || '',
  //     };
  //     getTaskRef(taskListRef, taskID).remove();
  //     taskListRef.push(updatedTask);
  //   }
  // };

  // deleteDrop = (event) => {
  //   event.preventDefault();

  //   const { taskListRef } = this.props;
  //   const dropTaskID = event.dataTransfer.getData('taskID');
  //   getTaskRef(taskListRef, dropTaskID).remove();

  //   handleDeleteDropCSS();
  // };

  render() {
    const { taskList } = this.props;

    const ToDoTasks = taskList.filter(task => (task.status === 'To Do'));
    const InProgressTasks = taskList.filter(task => (task.status === 'In Progress'));
    const DoneTasks = taskList.filter(task => (task.status === 'Done'));

    return (
      <Container fluid="true">
        <Row className="mt-3 justify-content-center">
          <Col md={10}>
            <MainInput />
          </Col>
        </Row>
        <Row className="mt-3 mb-3 mx-md-4 mx-lg-5" data-test="columnsRow">
          <Col md={4}>
            <TasksColumn
              title="To Do"
              sortedTasks={ToDoTasks}
              handleTaskDrop={this.handleTaskDrop}
            />
          </Col>
          <Col md={4}>
            <TasksColumn
              title="In Progress"
              sortedTasks={InProgressTasks}
              handleTaskDrop={this.handleTaskDrop}
            />
          </Col>
          <Col md={4}>
            <TasksColumn
              title="Done"
              sortedTasks={DoneTasks}
              handleTaskDrop={this.handleTaskDrop}
            />
          </Col>
        </Row>
        <Row className="justify-content-center">
          <Col
            className="delete-zone"
            id="delete-zone"
            xs={10}
            md={3}
            onDrop={e => this.deleteDrop(e)}
            // onDragOver={e => deleteDragByEvent(e, 'over')}
            // onDragLeave={e => deleteDragByEvent(e, 'leave')}
            onDragEnter={e => e.preventDefault()}
          >
            <img
              src={trash}
              alt="Delete"
              id="delete-can"
              className="delete-can"
              draggable="false"
            />
          </Col>
        </Row>
      </Container>
    );
  }
}

const mapStateToProps = ({ mainContainerReducer: { taskList, taskListRef } }) => ({
  taskList,
  taskListRef,
});

const mapDispatchToProps = dispatch => ({
  mainContainerActions: bindActionCreators(mainContainer, dispatch),
});

export { MainContainer as MainContainerComponent };
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MainContainer);
