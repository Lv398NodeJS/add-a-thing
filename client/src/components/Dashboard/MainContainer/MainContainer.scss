$delete-zone-color: #c2695f;

.loader {
  display: flex;
}

.loader * {
  margin: 0 auto;
}

.delete-zone {
  position: fixed;
  bottom: 4%;
  opacity: 0;
  z-index: 1000;
  height: 3.2rem;
  visibility: hidden;
  border-radius: 0.4rem;
  background-color: $delete-zone-color;
  transition: visibility 0s linear 300ms, opacity 300ms;
}

.delete-can {
  position: absolute;
  top: 30%;
  left: 50%;
  width: 1.5rem;
  transform: scale(1) translateX(-50%);
  transition: transform 150ms ease-in-out;
}

.shown {
  opacity: 1;
  visibility: visible;
  transition: visibility 0s ease-in-out, opacity 300ms;
}

.drag-in .delete-can {
  transform: scale(1.7) translateX(-30%);
}