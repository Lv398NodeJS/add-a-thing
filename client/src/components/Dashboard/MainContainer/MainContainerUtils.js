export const getTaskListAsArray = (snapValue = {}) => {
  const taskList = [];

  Object.keys(snapValue).forEach(task => (
    taskList.push({
      id: task,
      name: snapValue[task].name,
      priority: snapValue[task].priority,
      description: snapValue[task].description,
      status: snapValue[task].status,
      key: task,
    })));

  return taskList;
};

export const deleteDragByEvent = (event, type) => {
  event.preventDefault();
  if (type === 'over') {
    document.getElementById('delete-zone').classList.add('drag-in');
  } else if (type === 'leave') {
    document.getElementById('delete-zone').classList.remove('drag-in');
  }
};

export const handleDeleteDropCSS = () => {
  const fakeTask = document.getElementById('drag-avatar');
  document.getElementsByClassName('dragged-task')[0].remove();
  if (fakeTask != null) fakeTask.remove();
  document.getElementById('delete-zone').classList.remove('drag-in');
  const DELETE_DELAY = 150;
  setTimeout(() => document.getElementById('delete-zone').classList.remove('shown'), DELETE_DELAY);
};
