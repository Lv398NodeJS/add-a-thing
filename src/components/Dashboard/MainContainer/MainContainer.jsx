/* eslint-disable guard-for-in */
import React, { Component } from 'react';
import { Container } from 'react-bootstrap';
import ToDo from '../TasksColumns/ToDo';
import InProgress from '../TasksColumns/InProgress';
import Done from '../TasksColumns/Done';
import db from '../../../fire';
import MainInput from '../MainInput/MainInput';

export default class MainContainer extends Component {
  constructor() {
    super();
    this.state = {
      taskList: [],
      dashboardId: null,
    };
    this.addNewTask = this.addNewTask.bind(this);
  }

  componentDidMount() {
    const id = document.URL.split('/').pop();
    const taskListRef = db.database().ref(`dashboards/${id}/taskList`);
    taskListRef.on('value', (snapshot) => {
      const taskListSnap = snapshot.val();
      const newState = [];
      // eslint-disable-next-line no-restricted-syntax
      for (const task in taskListSnap) {
        newState.push({
          id: task,
          name: taskListSnap[task].name,
          description: taskListSnap[task].description,
          status: taskListSnap[task].status,
          // subTaskList: taskListSnap[task].subTaskList,
          key: task,
        });
      }
      this.setState(({
        dashboardId: id,
        taskList: newState,
      }));
    });
  }

  addNewTask = (inputData) => {
    if (inputData.length > 0) {
      this.setState(prevState => (
        {
          taskList: [...prevState.taskList, {
            name: inputData, description: '', status: 'To Do',
          }],
        }
      ));

      const id = document.URL.split('/').pop();
      const addTaskRef = db.database().ref(`dashboards/${id}/taskList`);
      const addTask = {
        name: inputData, description: '', status: 'To Do',
      };
      addTaskRef.push(addTask);
    }
  };

  render() {
    const ColumnsContainerStyle = {
      minHeight: '100vh',
      minWidth: '100vw',
      backgroundColor: 'rgb(247, 247, 247)',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'space-evenly',
      flexWrap: 'wrap',
    };

    const taskColumnStyle = {
      width: '400px',
      minHeight: '450px',
      color: 'rgb(194, 105, 95)',
      background: '#FFFFFF',
    };

    const { taskList } = this.state;

    const ToDoTasks = taskList.filter(task => (task.status === 'To Do'));
    const InProgressTasks = taskList.filter(task => (task.status === 'In Progress'));
    const DoneTasks = taskList.filter(task => (task.status === 'Done'));

    return (
      <Container className="ColumnsContainer" style={ColumnsContainerStyle}>
        <MainInput addNewTask={this.addNewTask} />
        <Container className="taskColumnContainer" style={taskColumnStyle}>
          <ToDo
            sortedTasks={ToDoTasks}
            taskList={taskList}
          />
        </Container>
        <Container className="taskColumnContainer" style={taskColumnStyle}>
          <InProgress
            sortedTasks={InProgressTasks}
            taskList={taskList}
          />
        </Container>
        <Container className="taskColumnContainer" style={taskColumnStyle}>
          <Done
            sortedTasks={DoneTasks}
            taskList={taskList}
          />
        </Container>
      </Container>
    );
  }
}
