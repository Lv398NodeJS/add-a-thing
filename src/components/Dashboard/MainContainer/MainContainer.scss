$main-container-color: rgb(247, 247, 247);
$task-column-color: rgb(255, 255, 255);

.mainContainer {
  display: flex;
  flex: 0 0 auto;
  justify-content: space-evenly;
  flex-wrap: wrap;
  padding-bottom: 10px;
  min-height: calc(100vh - 56px);
  min-width: 100vw;
  background-color: $main-container-color;
}

.tasksColumn {
  overflow: hidden;
  width: 400px;
  padding: 5px;
  height: 550px;
  background: $task-column-color;
}