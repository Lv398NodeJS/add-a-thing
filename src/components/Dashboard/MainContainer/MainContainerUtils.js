export const getTaskListAsArray = (snapValue = {}) => {
  const taskList = [];

  Object.keys(snapValue).forEach(task => (
    taskList.push({
      id: task,
      name: snapValue[task].name,
      priority: snapValue[task].priority,
      description: snapValue[task].description,
      status: snapValue[task].status,
      key: task,
    })));

  return taskList;
};

export const deleteDragOver = (event) => {
  event.preventDefault();
  document.getElementById('delete-zone').classList.add('drag-in');
};

export const deleteDragLeave = (event) => {
  event.preventDefault();
  document.getElementById('delete-zone').classList.remove('drag-in');
};

export const handleDeleteDropCSS = () => {
  const fakeTask = document.getElementById('drag-avatar');
  if (fakeTask != null) fakeTask.remove();

  document.getElementById('delete-zone').classList.remove('drag-in');
  setTimeout(() => document.getElementById('delete-zone').classList.remove('shown'), 150);
};
