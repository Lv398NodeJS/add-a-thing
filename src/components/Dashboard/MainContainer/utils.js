export const getTaskListAsArray = (snapValue = {}) => {
  const taskList = [];

  Object.keys(snapValue).forEach(task => (
    taskList.push({
      id: task,
      name: snapValue[task].name,
      priority: snapValue[task].priority,
      description: snapValue[task].description,
      status: snapValue[task].status,
      key: task,
    })));

  return taskList;
};
