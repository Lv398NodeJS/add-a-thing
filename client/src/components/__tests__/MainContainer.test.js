import React from 'react';
import { shallow } from 'enzyme';
import * as utils from '../Dashboard/MainContainer/MainContainerUtils';
import { MainContainerComponent } from '../Dashboard/MainContainer/MainContainer';

const mainContainer = shallow(<MainContainerComponent
  taskList={[]}
  mainContainerActions={{
    setTaskListRef: jest.fn(),
    fetchTaskList: jest.fn(),
  }}
/>);

describe('<MainContainer />', () => {
  it('should render and match snapshot', () => {
    expect(mainContainer).toMatchSnapshot();
  });
  it('should render row with columns', () => {
    const row = mainContainer.find('[data-test="columnsRow"]');
    expect(row.length).toBe(1);
  });
  it('should render 3 columns', () => {
    const columns = mainContainer.find('[data-test="columnsRow"]').children();
    expect(columns.length).toBe(3);
  });
});

describe('getTaskListAsArray()', () => {
  const obj = {
    '-Lcc0GCG-dvjcKH6BMEk': {
      description: 'Description for task',
      name: 'Create sub tasks',
      status: 'In Progress',
    },
  };

  it('should return an array', () => {
    expect(utils.getTaskListAsArray(obj)).toBeTruthy();
    expect(typeof utils.getTaskListAsArray(obj)).toEqual('object');
  });
});
