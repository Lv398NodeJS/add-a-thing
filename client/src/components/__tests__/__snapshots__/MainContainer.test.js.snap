// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainContainer /> should render and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainContainer
    mainContainerActions={
      Object {
        "fetchTaskList": [MockFunction] {
          "calls": Array [
            Array [
              "",
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "setTaskListRef": [MockFunction],
      }
    }
    taskList={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": Array [
        <ForwardRef(render)
          className="mt-3 justify-content-center"
        >
          <ForwardRef
            as="div"
            md={10}
          >
            <UNDEFINED />
          </ForwardRef>
        </ForwardRef(render)>,
        <ForwardRef(render)
          className="mt-3 mb-3 mx-md-4 mx-lg-5"
          data-test="columnsRow"
        >
          <ForwardRef
            as="div"
            md={4}
          >
            <UNDEFINED
              sortedTasks={Array []}
              title="To Do"
            />
          </ForwardRef>
          <ForwardRef
            as="div"
            md={4}
          >
            <UNDEFINED
              sortedTasks={Array []}
              title="In Progress"
            />
          </ForwardRef>
          <ForwardRef
            as="div"
            md={4}
          >
            <UNDEFINED
              sortedTasks={Array []}
              title="Done"
            />
          </ForwardRef>
        </ForwardRef(render)>,
        <ForwardRef(render)
          className="justify-content-center"
        >
          <ForwardRef
            as="div"
            className="delete-zone"
            id="delete-zone"
            md={3}
            onDragEnter={[Function]}
            onDrop={[Function]}
            xs={10}
          >
            <img
              alt="Delete"
              className="delete-can"
              draggable="false"
              id="delete-can"
              src="trash.svg"
            />
          </ForwardRef>
        </ForwardRef(render)>,
      ],
      "fluid": "true",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            as="div"
            md={10}
          >
            <UNDEFINED />
          </ForwardRef>,
          "className": "mt-3 justify-content-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <UNDEFINED />,
            "md": 10,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(MainInput)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(Row)",
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              as="div"
              md={4}
            >
              <UNDEFINED
                sortedTasks={Array []}
                title="To Do"
              />
            </ForwardRef>,
            <ForwardRef
              as="div"
              md={4}
            >
              <UNDEFINED
                sortedTasks={Array []}
                title="In Progress"
              />
            </ForwardRef>,
            <ForwardRef
              as="div"
              md={4}
            >
              <UNDEFINED
                sortedTasks={Array []}
                title="Done"
              />
            </ForwardRef>,
          ],
          "className": "mt-3 mb-3 mx-md-4 mx-lg-5",
          "data-test": "columnsRow",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": <UNDEFINED
                sortedTasks={Array []}
                title="To Do"
              />,
              "md": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleTaskDrop": undefined,
                "sortedTasks": Array [],
                "title": "To Do",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(TasksColumn)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": <UNDEFINED
                sortedTasks={Array []}
                title="In Progress"
              />,
              "md": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleTaskDrop": undefined,
                "sortedTasks": Array [],
                "title": "In Progress",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(TasksColumn)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": <UNDEFINED
                sortedTasks={Array []}
                title="Done"
              />,
              "md": 4,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleTaskDrop": undefined,
                "sortedTasks": Array [],
                "title": "Done",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(TasksColumn)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(Row)",
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            as="div"
            className="delete-zone"
            id="delete-zone"
            md={3}
            onDragEnter={[Function]}
            onDrop={[Function]}
            xs={10}
          >
            <img
              alt="Delete"
              className="delete-can"
              draggable="false"
              id="delete-can"
              src="trash.svg"
            />
          </ForwardRef>,
          "className": "justify-content-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <img
              alt="Delete"
              className="delete-can"
              draggable="false"
              id="delete-can"
              src="trash.svg"
            />,
            "className": "delete-zone",
            "id": "delete-zone",
            "md": 3,
            "onDragEnter": [Function],
            "onDrop": [Function],
            "xs": 10,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Delete",
              "className": "delete-can",
              "draggable": "false",
              "id": "delete-can",
              "src": "trash.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(Row)",
          "propTypes": undefined,
          "render": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": Array [
          <ForwardRef(render)
            className="mt-3 justify-content-center"
          >
            <ForwardRef
              as="div"
              md={10}
            >
              <UNDEFINED />
            </ForwardRef>
          </ForwardRef(render)>,
          <ForwardRef(render)
            className="mt-3 mb-3 mx-md-4 mx-lg-5"
            data-test="columnsRow"
          >
            <ForwardRef
              as="div"
              md={4}
            >
              <UNDEFINED
                sortedTasks={Array []}
                title="To Do"
              />
            </ForwardRef>
            <ForwardRef
              as="div"
              md={4}
            >
              <UNDEFINED
                sortedTasks={Array []}
                title="In Progress"
              />
            </ForwardRef>
            <ForwardRef
              as="div"
              md={4}
            >
              <UNDEFINED
                sortedTasks={Array []}
                title="Done"
              />
            </ForwardRef>
          </ForwardRef(render)>,
          <ForwardRef(render)
            className="justify-content-center"
          >
            <ForwardRef
              as="div"
              className="delete-zone"
              id="delete-zone"
              md={3}
              onDragEnter={[Function]}
              onDrop={[Function]}
              xs={10}
            >
              <img
                alt="Delete"
                className="delete-can"
                draggable="false"
                id="delete-can"
                src="trash.svg"
              />
            </ForwardRef>
          </ForwardRef(render)>,
        ],
        "fluid": "true",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              as="div"
              md={10}
            >
              <UNDEFINED />
            </ForwardRef>,
            "className": "mt-3 justify-content-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": <UNDEFINED />,
              "md": 10,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(MainInput)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(Row)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                as="div"
                md={4}
              >
                <UNDEFINED
                  sortedTasks={Array []}
                  title="To Do"
                />
              </ForwardRef>,
              <ForwardRef
                as="div"
                md={4}
              >
                <UNDEFINED
                  sortedTasks={Array []}
                  title="In Progress"
                />
              </ForwardRef>,
              <ForwardRef
                as="div"
                md={4}
              >
                <UNDEFINED
                  sortedTasks={Array []}
                  title="Done"
                />
              </ForwardRef>,
            ],
            "className": "mt-3 mb-3 mx-md-4 mx-lg-5",
            "data-test": "columnsRow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "div",
                "children": <UNDEFINED
                  sortedTasks={Array []}
                  title="To Do"
                />,
                "md": 4,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleTaskDrop": undefined,
                  "sortedTasks": Array [],
                  "title": "To Do",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(TasksColumn)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                },
                "displayName": "Col",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "div",
                "children": <UNDEFINED
                  sortedTasks={Array []}
                  title="In Progress"
                />,
                "md": 4,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleTaskDrop": undefined,
                  "sortedTasks": Array [],
                  "title": "In Progress",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(TasksColumn)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                },
                "displayName": "Col",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "div",
                "children": <UNDEFINED
                  sortedTasks={Array []}
                  title="Done"
                />,
                "md": 4,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleTaskDrop": undefined,
                  "sortedTasks": Array [],
                  "title": "Done",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.memo),
                  "WrappedComponent": [Function],
                  "compare": null,
                  "displayName": "Connect(TasksColumn)",
                  "type": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                },
                "displayName": "Col",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(Row)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              as="div"
              className="delete-zone"
              id="delete-zone"
              md={3}
              onDragEnter={[Function]}
              onDrop={[Function]}
              xs={10}
            >
              <img
                alt="Delete"
                className="delete-can"
                draggable="false"
                id="delete-can"
                src="trash.svg"
              />
            </ForwardRef>,
            "className": "justify-content-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": <img
                alt="Delete"
                className="delete-can"
                draggable="false"
                id="delete-can"
                src="trash.svg"
              />,
              "className": "delete-zone",
              "id": "delete-zone",
              "md": 3,
              "onDragEnter": [Function],
              "onDrop": [Function],
              "xs": 10,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Delete",
                "className": "delete-can",
                "draggable": "false",
                "id": "delete-can",
                "src": "trash.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(Row)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
